catalyst/loggers/wandb.py

```diff
@@ -0,0 +1,180 @@
+    ) -> None:
+        self.project = project
+        self.name = name
+        self.config = config
+        self.entity = entity
+        self.run = wandb.init(
+            project=self.project, name=self.name, config=self.config, entity=self.entity
+        )
+
+    def _log_metrics(self, metrics: Dict[str, float], step: int, loader_key: str, suffix=""):
+        for key, value in metrics.items():
+            self.run.log({f"{key}/{loader_key}{suffix}": value}, step=step)
+
+    def log_metrics(
+        self,
+        metrics: Dict[str, Any],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs batch and epoch metrics to wandb."""
+        if scope == "batch":
+            metrics = {k: float(v) for k, v in metrics.items()}
+            self._log_metrics(
+                metrics=metrics, step=global_batch_step, loader_key=loader_key, suffix="/batch"
+            )
+        elif scope == "epoch":
+            for loader_key, per_loader_metrics in metrics.items():
+                self._log_metrics(
+                    metrics=per_loader_metrics,
+                    step=global_epoch_step,
+                    loader_key=loader_key,
+                    suffix="/epoch",
+                )
+
+    def log_image(
+        self,
+        tag: str,
+        image: np.ndarray,
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs image to the logger."""
+        self.run.log(
+            {f"{tag}_scope_{scope}_epoch_{global_epoch_step}.png": wandb.Image(image)},
+            step=global_epoch_step,
+        )
+
+    def log_hparams(
+        self,
+        hparams: Dict,
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        stage_key: str = None,
+    ) -> None:
+        """Logs hyperparameters to the logger."""
+        self.run.config.update(hparams)
```

Comment:
would that update the config on the Wandb side?

Reply:
Yes. But this is not recommended. I'll comment it out. config when updated, should ideally only add new keys and should avoid updating the old keys which isn't guaranteed here. That's what I wanted to discuss regarding config. The recommended way here would be to pass config hparams in the construntor as:
`logger['wandb'] = WandbLogger(project='name, config=self.harams)`

But anyway neither of these are updating any values in config as it stays empty when running test_finetune2.py

Reply:
The WandbLogger looks great!
Let's resolve this issue and merge it to the master branch.
I think, we could return `self.run.config.update(hparams)` to `log_hparams` and update the docs with the ideal use case scenario: pass the experiment hyperparameter through `log_hparams` rather than `init`.

Reply:
Done :slightly_smiling_face: